modules/data-entry/src/main/frontend/src/vocabQuery/VocabularyTree.jsx

```diff
@@ -145,14 +184,20 @@ function VocabularyTree(props) {
       withCloseButton
       open={open}
       ref={browserRef}
-      onClose={onClose}
+      onClose={() => {onClose(selectedTerms, removedTerms); setRemovedTerms([]);}}
       className={classes.dialog}
       classes={{
         paper: classes.dialogPaper,
         root: classes.infoDialog
       }}
       {...rest}
     >
+      { allowTermSelection &&
+        <div className={classes.selectionContainer}>
+          <Typography variant="body2" component="span">{questionDefinition?.text}:</Typography>
+          { selectedTerms?.filter(i => i[LABEL_POS]).map(s => <Chip key={s[VALUE_POS]} variant="outlined" size="small" label={s[LABEL_POS]} onDelete={() => onRemoveOption(...s)} color="primary" className={classes.selectionChips}/>) }
```

Comment:
Let's make this easier to read:
```suggestion
          { selectedTerms?.filter(i => i[LABEL_POS]).map(s =>
             <Chip
               key={s[VALUE_POS]}
               variant="outlined"
               size="small"
               color="primary"
               label={s[LABEL_POS]}
               onDelete={() => onRemoveOption(...s)}
               className={classes.selectionChips}
             />
           )}
```